Source/Spring2022_Capstone/Enemies/BaseEnemy.h

```diff
@@ -0,0 +1,46 @@
+// Created by Spring2022_Capstone team
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "GameFramework/Character.h"
+#include "BaseEnemy.generated.h"
+
+class UHealthComponent;
+
+UCLASS(Abstract)
+class SPRING2022_CAPSTONE_API ABaseEnemy : public ACharacter
+{
+	GENERATED_BODY()
+
+public:
+	// Sets default values for this character's properties
+	ABaseEnemy();
+
+protected:
+	// Called when the game starts or when spawned
+	virtual void BeginPlay() override;
+
+	UPROPERTY(VisibleAnywhere, Category = "Components", meta = (AllowPrivateAccess = true))
+	UStaticMeshComponent *WeaponMesh;
+	UPROPERTY(VisibleAnywhere, Category = "Components", meta = (AllowPrivateAccess = true))
+	USceneComponent *ProjectileSpawnPoint;
+
+	UPROPERTY(EditAnywhere, BlueprintReadOnly, Category = Input, meta = (AllowPrivateAccess = "true"))
+	UHealthComponent *HealthComponent;
+
+	virtual void Attack();
+
+public:
+	// Called every frame
+	virtual void Tick(float DeltaTime) override;
+
+	// Called to bind functionality to input
+	virtual void SetupPlayerInputComponent(class UInputComponent *PlayerInputComponent) override;
```

Comment:
Do we really need this, I assume enemies would not have any input bound to them.